### PR TITLE
AAP-30617: Introduce AACSAPIView to handle the generic aspect of the view

### PR DESCRIPTION
**Pull request by goneri**
Introduce the `AACSAPIView` class to handle the generic parts of the view. For now, we use it only for the `Explanation` view and it will be extended to the other views later.

The class is in charge of:

- the Schema1 event creation and emission
- the exception handling
- the parameter handling (not in this PR, stay turned!)

### CODE REVIEW

File: ansible_ai_connect/ai/api/views.py [hunk 1]
```diff
@@ -160,6 +161,64 @@
...
+        response = super().handle_exception(exc)
+        self.event.set_exception(exc)
+        return response
+
+    def finalize_response(self, request, response, *args, **kwargs):
+        print("finalize_response")
+        response = super().finalize_response(request, response, *args, **kwargs)
+
+        try:
+            model_meta_data: MetaData = apps.get_app_config("ai").get_model_pipeline(MetaData)
```
Comment (manstis) [inline ansible_ai_connect/ai/api/views.py:211]: :partying_face:

File: ansible_ai_connect/ai/api/views.py [hunk 2]
```diff
@@ -635,7 +694,7 @@ def write_to_segment(
         send_segment_event(event, "contentmatch", user)
 
 
-class Explanation(APIView):
+class Explanation(AACSAPIView):
```
Comment (manstis) [inline ansible_ai_connect/ai/api/views.py:697]: Out of interest, what does `AACS` stand for?

File: ansible_ai_connect/ai/api/views.py [hunk 3]
```diff
@@ -160,6 +161,64 @@
...
+            if isinstance(exc, original_class):
+                exc = new_class(cause=exc)
+                break
+        logger.error(f"{type(exc)}: {traceback.format_exception(exc)}")
+        response = super().handle_exception(exc)
+        self.event.set_exception(exc)
+        return response
+
+    def finalize_response(self, request, response, *args, **kwargs):
+        print("finalize_response")
```
Comment (manstis) [inline ansible_ai_connect/ai/api/views.py:207]: Did you want to keep this?

**Review by manstis — DISMISSED** [review]
Nice improvement. LGTM :+1:

File: ansible_ai_connect/ai/api/views.py (hunk 2 above)
Comment (goneri) [reply ansible_ai_connect/ai/api/views.py:697]: Ansible-Ai-Connect-Service obviously :-)

File: ansible_ai_connect/ai/api/views.py (hunk 3 above)
Comment (goneri) [reply ansible_ai_connect/ai/api/views.py:207]: cough cough, maybe not.

**Comment by goneri** [comment]
> Nice improvement. LGTM 👍

Thanks, but first I need to get the CI green.

File: ansible_ai_connect/ai/api/views.py (hunk 2 above)
Comment (manstis) [reply ansible_ai_connect/ai/api/views.py:697]: LOL. OK, we've been using `AAIC` in the Lightspeed Operator.

**Review by manstis — APPROVED** [review]
LGTM :+1:

**Review by jameswnl — APPROVED** [review]

**Pull request merged**